ripple/BRRippleTransaction.c

```diff
@@ -318,10 +318,11 @@ rippleTransactionSerializeAndSign(BRRippleTransaction transaction, BRKey * priva
     // Serialize and sign the tx bytes
     BRRippleSerializedTransaction serializedBytes = rippleTransactionSerializeImpl (transaction, 0, 0);
     BRRippleSignature sig = signBytes(privateKey, serializedBytes->buffer, serializedBytes->size);
-    free(serializedBytes); // Free the serialized bytes since we have to reserialize with signature below
```

Comment:
I guess I missed the fact that this structure had a member that was also allocated memory

Reply:
I know this is not part of your change - but I do notice that down on line 335 we are dereferencing transaction->signedBytes without checking if the pointer is valid

Reply:
Yes, that needs to be fixed!

Reply:
Like this `return (NULL == transaction->signedBytes ? 0 : transaction->signedBytes->size);`  Okay?